src/snakemake_software_deployment_plugin_conda/__init__.py

```
@@ -485,12 +499,28 @@ async def deploy(self) -> None:
+            try:
+                self.run_cmd(
+                    self.decorate_shellcmd(f"sh {self.spec.post_deploy_script.cached}"),
+                    check=True,
+                    stdout=sp.PIPE,
+                    stderr=sp.STDOUT,
+                )
+            except sp.CalledProcessError as e:
+                raise WorkflowError(
+                    "Failed to run post-deploy script "
+                    f"{self.spec.post_deploy_script.path_or_uri}: {e.stdout.decode()}"
+                )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)